### ISSUE TITLE
calling setTimeout throws a maximum call stack exception

### ISSUE THREAD

**Issue by dylanb** [angular/angular#1050]
1. Load any Angular 2 application
2. Open the Chrome developer tools
3. Switch to the console
4. Enter `setTimeout(function () {console.log('timeout');}, 0);`
5. Note the exception (reproduced below) for the maximum stack exception

This is simply a very easy way to reproduce a problem that occurs in real applications/components that need to use timeouts inside event handlers. Needs to be fixed.

```
Uncaught RangeError: Maximum call stack size exceeded es 6-shim.js:29889
run                                                   es 6-shim.js:29866
zoneBoundFn                                           es6-shim.js:29940
arguments.(anonymous function)                        es6-shim.js:29873
(anonymous function)                                  es6-shim.js:29889
run                                                   es6-shim.js:29866
zoneBoundFn                                           es6-shim.js:29940
arguments.(anonymous function)                        es6-shim.js:29873
(anonymous function)                                  es6-shim.js:29889
run                                                   es6-shim.js:29866
```

**Comment by Invis1ble** [comment]
The same issue with `setInterval` and `requestAnimationFrame`

**Comment by wesleycho** [comment]
If you load just zone.js and not angular 2, does this occur still?

**Comment by dylanb** [comment]
@wesleycho tried to figure out how to just load zone.js, but I am using the quickstart repo and it seems like `vm_turn_zone.js` pulls a bunch of dependencies along with it.

How do I test just the zone code?

**Comment by Invis1ble** [comment]
I'm not sure is fiddle is correct, but seems like using just zone.js is not triggering this error
http://jsfiddle.net/d6k05n3z/

**Comment by dylanb** [comment]
@Invis1ble looks like you are correct

**Comment by pkozlowski-opensource** [comment]
Bumping into this one as well, here is the reproduce scenario: http://plnkr.co/edit/fvviNUSb9mQ4Ky1S4MCf?p=preview

Related issue on zone.js side: https://github.com/angular/zone.js/issues/57

**Comment by formula1** [comment]
Glad to see I'm not the only one

**Comment by naomiblack** [comment]
Related: https://github.com/angular/angular/issues/1083

**Comment by vicb** [comment]
Fixed since last 0.4 release of zone.js

**Comment by dylanb** [comment]
Nice - I can confirm this works
